File: app/src/main/java/com/gnoemes/shikimoriapp/entity/rates/domain/UserRate.java
package com.gnoemes.shikimoriapp.entity.rates.domain;

import android.support.annotation.Nullable;

import com.gnoemes.shikimoriapp.entity.main.domain.Constants;

import java.io.Serializable;

public class UserRate implements Serializable {

    private long id;
    private RateStatus status;
    @Nullable
    private String score;
    @Nullable
    private String text;
    @Nullable
    private String episodes;
    @Nullable
    private String rewatches;

    public UserRate(long id, RateStatus status, @Nullable String score,
                    @Nullable String text, @Nullable String episodes, @Nullable String rewatches) {
        this.id = id;
        this.score = score;
        this.status = status;
        this.text = text;
        this.episodes = episodes;
        this.rewatches = rewatches;
    }

    public UserRate(long id, RateStatus status) {
        this.id = id;
        this.status = status;
    }

    public long getId() {
        return id;
    }

    @Nullable
    public String getScore() {
        return score;
    }

    public RateStatus getStatus() {
        return status;
    }

    @Nullable
    public String getText() {
        return text;
    }

    @Nullable
    public String getEpisodes() {
        return episodes;
    }

    @Nullable
    public String getRewatches() {
        return rewatches;
    }

    public static UserRate getStartWatching() {
        return new UserRate(Constants.NO_ID, RateStatus.WATCHING);
    }
}

File: app/src/main/java/com/gnoemes/shikimoriapp/entity/rates/domain/UserRate.kt
package com.gnoemes.shikimoriapp.entity.rates.domain

import android.os.Parcelable
import com.gnoemes.shikimoriapp.entity.app.domain.Type
import com.gnoemes.shikimoriapp.entity.main.domain.Constants
import kotlinx.android.parcel.Parcelize
import org.joda.time.DateTime

@Parcelize
data class UserRate(
        val id: Long? = null,
        val userId: Long? = null,
        val targetId: Long? = null,
        val targetType: Type? = null,
        val score: Double? = null,
        val status: RateStatus? = null,
        val rewatches: Int? = null,
        val episodes: Int? = null,
        val volumes: Int? = null,
        val chapters: Int? = null,
        val text: String? = null,
        val textHtml: String? = null,
        val dateCreated: DateTime? = null,
        val dateUpdated: DateTime? = null
) : Parcelable {
    //for java
    companion object {
        @JvmStatic
        fun getStartWatching(): UserRate = UserRate(Constants.NO_ID, status = RateStatus.WATCHING)
    }
}